mpack/ambari_mpack.py

```
@@ -0,0 +1,300 @@
+        data = '{"Clusters":{"desired_config":['
+        data = data + "{" + f'"type": "{config_type}", "properties" : {properties}' + "},"
+        data = data[:-1] + ']}}'.replace('\'', '"')
+        self.api_client.put(self.api_client.clusters(self.cluster_name).url,
+                            content_type="application/json",
+                            data=data)
+        self.changed = True
+
+    def _add_service_component_to_service(self, component):
+        component_url = \
+            self.api_client.clusters(self.cluster_name).services(self.service_name).components(component).url
+        self.api_client.post(component_url, content_type="application/json")
+        self.changed = True
+
+    def _create_service(self):
+        self.api_client.post(self.api_client.clusters(self.cluster_name).services(self.service_name).url,
+                             content_type="application/json")
+        self._update()
+        self.changed = True
+
+    def _restart_stale_config_components(self):
+        self.api_client.clusters(self.cluster_name).restart_stale_config_components()
+        self.api_client.post(self.api_client.clusters(self.cluster_name).requests.url, data={
+            "RequestInfo": {
+                "context": ('Restart all required services'),
+                "operation_level": "host_component",
+                "command": "RESTART"
+            },
+            "Requests/resource_filters": [{
+                "hosts_predicate": "HostRoles/stale_configs=true&HostRoles/cluster_name=" + self.cluster_name
+            }],
+        })
+        self.changed = True
+
+    def deleted(self):
+        if self.status != "ABSENT":
+            self._stop_service()
+            self._delete_service()
+
+    def _stop_service(self):
+        try:
+            self.api_client.clusters(self.cluster_name).services(self.service_name).stop().wait()
+            self.changed = True
+        except:
```

Using a bare except is not recommended.
"except Exception", at least

---

done